fe1-web/store/reducers/SocialReducer.ts

```diff
@@ -86,21 +86,26 @@ const socialSlice = createSlice({
 
         const store = state.byLaoId[laoId];
 
-        if (!store.byId[chirp.id]) {
+        // add the chirp to byId or update the chirp state if it's already in byId
+        // if the chirp has not been added (didn't receive a delete request before)
+        // or if the addChirp and deleteChirp of this chirp.id are sent by different sender
+        // in this case we trust the sender of addChirp and keep the chirp as not deleted
+        if (!(store.byId[chirp.id]
+          && store.byId[chirp.id].sender === chirp.sender)) {
           store.byId[chirp.id] = chirp;
+        }
 
-          const insertIdxInAll = findInsertIdx(
-            store.allIdsInOrder, store.byId, chirp.time,
-          );
-          store.allIdsInOrder.splice(insertIdxInAll, 0, chirp.id);
-
-          if (!state.byLaoId[laoId].byUser[chirp.sender]) {
-            store.byUser[chirp.sender] = [chirp.id];
-          } else {
-            const senderChirps = store.byUser[chirp.sender];
-            const insertIdxInUser = findInsertIdx(senderChirps, store.byId, chirp.time);
-            senderChirps.splice(insertIdxInUser, 0, chirp.id);
-          }
+        const insertIdxInAll = findInsertIdx(
+          store.allIdsInOrder, store.byId, chirp.time,
+        );
+        store.allIdsInOrder.splice(insertIdxInAll, 0, chirp.id);
+
+        if (!state.byLaoId[laoId].byUser[chirp.sender]) {
+          store.byUser[chirp.sender] = [chirp.id];
+        } else {
+          const senderChirps = store.byUser[chirp.sender];
+          const insertIdxInUser = findInsertIdx(senderChirps, store.byId, chirp.time);
+          senderChirps.splice(insertIdxInUser, 0, chirp.id);
```

Comment:
Good job on handling the out-of-order messages.
I believe you've changed the adding logic here though. You're *always* going to modify `allIdsInOrder`, whereas before you were doing it only when adding. Is this on purpose ?

Reply:
Yes, because we use `allIdsInOrder` to display all chirps (including the deleted ones) in the right order. 
For out-of-message messages, there're two possible cases: 
1. same senders: we keep the state in `byId` as deleted, but still add it to `allIdsInOrder` to show it as a deleted chirp using the `addChirp` time.
2. different senders: the sender doesn't want to delete the chirp, we do the same things as for a normal add.
I will add a comment here to make it clear.